dotCMS/src/main/java/com/dotcms/rest/api/v1/JobQueueManagerHelper.java

```diff
@@ -0,0 +1,82 @@
+package com.dotcms.rest.api.v1;
+
+import com.dotcms.jobs.business.api.JobProcessorScanner;
+import com.dotcms.jobs.business.api.JobQueueManagerAPI;
+import com.dotcms.jobs.business.processor.JobProcessor;
+import com.dotcms.jobs.business.processor.Queue;
+import com.dotmarketing.util.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.lang.reflect.Constructor;
+import java.util.List;
+
+@ApplicationScoped
+public class JobQueueManagerHelper {
+
+    private JobProcessorScanner scanner;
+
+    @Inject
+    public JobQueueManagerHelper(JobProcessorScanner scanner) {
+        this.scanner = scanner;
+    }
+
+    public JobQueueManagerHelper() {
```

Comment:
why do you need an empty constructor?

Reply:
CDI needs an empty constructor when it creates a proxy by subclassing a class because Java requires the subclass to call the parent class’s constructor. However, if the class implements an interface, no empty constructor is needed, since the proxy won't rely in that case on subclassing.